src/sentry/snuba/metrics/extraction.py

```diff
@@ -1136,6 +1137,15 @@ def __init__(
 
         # Removes field if passed in selected_columns
         self.groupbys = [groupby for groupby in groupbys or () if groupby != field]
+        # In the previous version we were not including the environment tag in the hash.
+        # Including it in the groupsby will include it in the query hash
+        if (
+            self.spec_type == MetricSpecType.DYNAMIC_QUERY
+            and "environment" not in self.groupbys
+            and self.spec_version.flags == {"include_environment_tag"}
+        ):
+            # XXX: Should this always be included?
+            self.groupbys.append("environment")
```

Comment:
Yep, this is looking right, and yes, so long as it's `DYNAMIC_QUERY` environment should _always_ be there so people can change environment on the fly in their dashboards.